packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -861,6 +886,50 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
           [queryStateResults, querySubscriptionResults]
         )
       },
+      useUnstable_SuspenseQuery(arg, options) {
+        const querySubscriptionResults = useQuerySubscription(arg, options)
+        const queryStateResults = useQueryState(arg, {
+          selectFromResult: isSkippedQuery(arg, options)
+            ? undefined
+            : noPendingQueryStateSelector,
+          ...options,
+        })
+
+        // We do not suspend if a query is skipped:
+        // @see https://github.com/vercel/swr/pull/357#issuecomment-627089889
+        if (!isSkippedQuery(arg, options)) {
+          if (
+            queryStateResults.isLoading ||
+            (queryStateResults.isFetching && options?.suspendOnRefetch)
+          ) {
+            const pendingPromise = api.util.getRunningOperationPromise<any>(
+              name,
+              arg as unknown as string
+            )
+
+            if (pendingPromise) {
+              throw pendingPromise
+            }
+          } else if (
+            queryStateResults.isError &&
+            !queryStateResults.isFetching
```

Comment:
I think this is interesting behavior. Not sure how everyone would expect this to behave. 

i would think that if there is an error, it will *always* throw an error. and once the user resets the error boundary and thus remounts the component that had the suspended query, it will refetch again